.pre-commit-config.yaml

```diff
@@ -0,0 +1,55 @@
+# Copyright (C) 2022 Dremio Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+default_language_version:
+  python: python3
+
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v3.2.0
+  hooks:
+    - id: trailing-whitespace
+    - id: end-of-file-fixer
+    - id: check-yaml
+    - id: check-added-large-files
+    - id: check-ast
+    - id: check-merge-conflict
+    - id: debug-statements
+    - id: detect-private-key
+- repo: https://github.com/psf/black
+  rev: 22.10.0
+  hooks:
+  - id: black
+    additional_dependencies: ['click==8.0.4']
+    args:
+    - "--target-version=py310"
+  - id: black
+    alias: black-check
```

Comment:
Good question. The `stages: [manual]` means:

> "An additional manual stage is available for one off execution that won't run in any hook context. This special stage is useful for taking advantage of pre-commit's cross-platform / cross-language package management without running it on every commit. Hooks confined to stages: [manual] can be executed by running pre-commit run --hook-stage manual [hookid]."

I don't know how useful it would be us to keep that `manual` stage in, so I will remove that.

https://pre-commit.com/#config-stages